python/feldera/pipeline_builder.py

```diff
@@ -0,0 +1,155 @@
+from typing import Self
+
+from feldera.rest.feldera_client import FelderaClient
+from feldera.rest.pipeline import Pipeline as InnerPipeline
+from feldera.pipeline import Pipeline
+from feldera.enums import CompilationProfile
+from feldera.runtime_config import RuntimeConfig, Resources
+from feldera.rest.errors import FelderaAPIError
+
+
+class PipelineBuilder:
+    """
+    A builder for creating a Pipeline in Feldera.
+
+    :param client: The `.FelderaClient` instance
+    :param name: The name of the pipeline
+    :param description: The description of the pipeline
+    :param sql: The SQL code of the pipeline
+    :param compilation_profile: The compilation profile to use
+    :param runtime_config: The runtime config to use
+    """
+
+    def __init__(
+        self,
+        client: FelderaClient,
+        name: str = None,
+        sql: str = "",
+        description: str = "",
+        compilation_profile: CompilationProfile = CompilationProfile.OPTIMIZED,
+        runtime_config: RuntimeConfig = RuntimeConfig(resources=Resources()),
+
+    ):
+        self.client: FelderaClient = client
+        self.name: str | None = name
+        self.description: str = description
+        self.sql: str = sql
+        self.compilation_profile: CompilationProfile = compilation_profile
+        self.runtime_config: RuntimeConfig = runtime_config
+
+    def with_sql(self, sql: str) -> Self:
```

Comment:
Could be useful in notebooks. For example, the Hopsworks demo notebook 1.

Reply:
In https://github.com/feldera/feldera/blob/main/demo/project_demo11-Hopsworks/1_feature_pipeline.ipynb I don't see `pipeline` used anywhere up until just setting the SQL? It could have also just been created later.

Reply:
Step 1.2 can just be "Formulate Feldera SQL" and Step 1.3 "Create and run Feldera pipeline"